superphy/version-1
# Final delete

## Instructions for installing & testing branch

Changes were made to the DB. You will need to implement these changes in your DB.

Note: you will need a test DB from a run of the t/pipeline.t.  The pipeline.t script creates a copy of the genodo DB and installs a single user-submitted genome. You can use the user-submitted genome to play around with the edit function and delete function. The test DBs have names like testodo12345678.  To substitute a test DB for genodo, make a copy of the genodo.cfg file, replacing

  [db]
  ...
  name=genodo
  ...

with the name of the test DB.

TESTING INSTRUCTIONS:

1) Apply changes to the DB (this will be the testodo12345678 database if you are using it to do the testing)
- Make a backup of your DB
- In psql:
  DROP TABLES feature_group;
  DROP TABLE private_feature_group;
  DROP TABLE genome_group;
  DROP TABLE group_category;
  \q
- Insert new tables:
  psql dbname < Database/db_dumps/create_user_groups_tables.sql
- Reload meta-data groups:
  perl Data/initialize_standard_strain_groups.pl --config yourconfigfile.cfg
- Build missing indices
  psql dbname < Database/db_dumps/add_missing_indices.sql
- Add deleted_upload table
  psql dbname < Database/db_dumps/create_deleted_upload_table.sql

2) Run edit test script
- This test script creates a new temporary postgres DB from scratch and loads 60 genomes with associated meta-data. It requires Test::PostgreSQL to be installed.
- create directory:
  t/sandbox
  if it doesn't exist.
- run from the repo root:
  prove -lv t/edit-genome.t
  Note: the script sometimes fails because psql does not exit cleanly (you will see a message: Pg refused to die gracefully; killing it violently). Verify that all tests passed ok and ignore the Test::PostgreSQL issues.

3) Edit UI testing
The test script does not do any UI testing.  If you have a test DB available as mentioned above, you can edit the uploaded genome on the webserver.
- Make sure that the config file that is loaded by the webserver is accessing the DB with the uploaded genome. You can set the config file on the webserver through the environment variable: SUPERPHY_CONFIGFILE
- Login as 'testbot'.  This is the user that pipeline.t uses to upload the genome.
- Access edit form from menu: My Data > Modify or delete a genome > click on 'Modify' link next genome
- Go crazy in the edit form
- The webserver submits the edit form changes to a queue but does not actually implement the edits. The process Data/genodo_pipleine.pl (this is run ever 15 or so minutes on Giant) checks the queue and actually installs the changes in the DB:
  perl Data/genodo_pipeline.pl --config yourconfigfile.cfg
- Check 'view' or 'modify' pages for the genome to verify if changes are present.

4) Run delete test script
- This test makes a copy of an existing DB and deletes a random genome from the public tables and another genome from the private tables. It requires a private genome has been uploaded to the DB.  For this test it is recommended that you use one of the test DBs produced by the pipeline.t test script mentioned above (e.g. testodo12345678), which will contain a private genome uploaded by the 'testbot' user.
- Using DB with private genome as template, run from repo root:
  SUPERPHY_CONFIGFILE=../config/genodo.cfg TEMPLATE=testodo12345678 DBUSER=dbuser prove -lv t/delete-genome.t
